src/main/java/net/fabricmc/mappingio/tree/MappingTree.java

```diff
@@ -216,7 +216,8 @@ default MethodMapping getMethod(String name, String desc, int namespace) {
 			for (MethodMapping method : getMethods()) {
 				if (!name.equals(method.getDstName(namespace))) continue;
 				String mDesc;
-				if (desc != null && (mDesc = method.getDesc(namespace)) != null && !desc.equals(mDesc)) continue;
+				if (desc != null && (mDesc = method.getDesc(namespace)) != null && !desc.equals(mDesc)
+						&& !(desc.endsWith(")") && mDesc.startsWith(desc))) continue;
```

Comment:
No split inline if please, may as well split it all like so:

```suggestion

				if (desc != null 
						&& (mDesc = method.getDesc(namespace)) != null 
						&& !desc.equals(mDesc)
						&& !(desc.endsWith(")") 
						&& mDesc.startsWith(desc))) {
					continue;
				}
```